tools/lsp/ui/component-list.slint

```diff
@@ -46,25 +54,39 @@ export component ComponentList {
                         }
                     }
 
-                    for ci in cli.components: TouchArea {
+                    for ci in cli.components: component := TouchArea {
                         private property <length> drop-x: self.absolute-position.x + self.mouse-x - root.preview-area-position-x;
                         private property <length> drop-y: self.absolute-position.y + self.mouse-y - root.preview-area-position-y;
                         private property <bool> on-drop-area:
                             drop-x >= 0 && drop-x <= root.preview-area-width && drop-y >= 0 && drop-y <= root.preview-area-height;
-                        private property <bool> can-drop-here: root.can-drop(ci, drop-x, drop-y, on-drop-area);
+                        private property <bool> can-drop-here: !self.data.is-currently-shown && root.can-drop(self.data.name, drop-x, drop-y, on-drop-area);
+                        private property <ComponentItem> data: ci;
+                        private property <bool> edit-component-name;
 
                         enabled: root.preview-visible;
-                        height: name.preferred-height + 10px;
                         width: 100%;
-                        name := Text {
-                            text: ci;
+                        if self.data.is-user-defined && self.edit-component-name: LineEdit {
+                            height: self.min-height;
+                            text: component.data.name;
+                            horizontal-alignment: center;
+                            accepted(name) => {
+                                root.rename_component(component.data.name, component.data.defined_at, name);
+                                component.edit-component-name = false;
+                            }
+                        }
+                        if !(self.data.is-user-defined && self.edit-component-name): Text {
+                            height: self.min-height;
+                            text: component.data.name;
                         }
 
                         pointer-event(event) => {
                             if (self.can-drop-here && event.kind == PointerEventKind.up && event.button == PointerEventButton.left) {
-                                root.drop(ci, drop-x, drop-y);
+                                root.drop(self.data.name, drop-x, drop-y);
                             }
                         }
+                        clicked => {
+                            self.edit-component-name = self.data.is-user-defined && !self.edit-component-name;
```

Comment:
I don't think clicking should change the component name just like that. I think clicking should focus the editor to the element. 

To rename, it would be more like right-click and then have a popup menu with rename. Or maybe some keyboard shortcut

Reply:
I agree. There are OS-specific patterns for this, which I think we should re-use instead of introducing new patterns. Here's what I see on macOS and Windows to rename for example a file in the file manager:

1. Click the name to select it. It's visible that the name is selected.
2. Click again to rename it. (see https://devblogs.microsoft.com/oldnewthing/20081110-00/?p=20263 for caveats)
3. On Windows, the rename on the selected item can also be initiated with the F2 shortcut.